File: baekjoon_all/09000+/boj_9375.py
# Solve 2023-10-09

import sys

input = lambda : sys.stdin.readline().rstrip()

for _ in range(int(input())):
    clothes_type_cnt = dict()

    for _ in range(int(input())):
        clothes_name, clothes_type = input().split()

        if clothes_type in clothes_type_cnt:
            clothes_type_cnt[clothes_type] += 1
        else:
            clothes_type_cnt[clothes_type] = 1

    ans = 1

    for clothes_type in clothes_type_cnt:
        ans *= clothes_type_cnt[clothes_type] + 1

    print(ans - 1)

File: baekjoon_all/09000+/boj_9375.cpp
// Solve 2022-06-08
// Update 2023-10-09

#include <bits/stdc++.h>
using namespace std;

#define FASTIO ios_base::sync_with_stdio(false);cin.tie(NULL); // boj_15552.cpp
#define SETPRECISION(n) cout << fixed;cout.precision(n); // boj_1008.cpp
#define SIZE(v) (int)v.size()
#define ALL(v) v.begin(),v.end()
using ll = long long;

int main() {
    FASTIO;

    int t;
    cin >> t;

    for (int ti = 0; ti < t; ti++) {
        int n;
        cin >> n;

        unordered_map<string, int> clothes_type_cnt;

        for (int i = 0; i < n; i++) {
            string clothes_name, clothes_type;
            cin >> clothes_name >> clothes_type;

            clothes_type_cnt[clothes_type]++;
        }

        int ans = 1;

        for (auto it = clothes_type_cnt.begin(); it != clothes_type_cnt.end(); it++) {
            ans *= it->second + 1;
        }

        cout << --ans << '\n';
    }

    return 0;
}
